File: backend/src/main/java/dev/voroby/client/dto/ChatMessage.java
package dev.voroby.client.dto;

public record ChatMessage(
        long id,
        long chatId,
        boolean privateChat,
        String messageText,
        String photoPreview,
        String date,
        String editDate,
        String senderInfo,
        String senderPhoto,
        boolean isCurrentUser,
        boolean canBeDeletedForAllUsers,
        boolean canBeDeletedOnlyForSelf
) {}

File: desktop/src/main/kotlin/chat/dto/ChatMessage.kt
package chat.dto

data class ChatMessage(
    val id: Long,
    val chatId: Long,
    val privateChat: Boolean,
    val textContent: Content.TextContent,
    val encodedContent: Content.EncodedContent?,
    val urlContent: Content.UrlContent?,
    val date: String,
    val editDate: String,
    val senderInfo: String,
    val senderPhoto: String,
    val isCurrentUser: Boolean,
    val canBeDeletedForAllUsers: Boolean,
    val canBeDeletedOnlyForSelf: Boolean
)
